.github/workflows/pr-coverage.yml

```diff
@@ -0,0 +1,73 @@
+name: Test Coverage
+on: ["push", "pull_request"]
+
+concurrency:
+  group: ${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+env:
+  REALM_DISABLE_ANALYTICS: 1
+  MOCHA_TIMEOUT: 60000 # 1 minute
+  LONG_TIMEOUT: 300000 # 5 minutes
+  SERVER_TAG: latest
+  # REALM_BASE_URL: ${{ secrets.REALM_QA_BASE_URL }}
+  # REALM_PUBLIC_KEY: ${{ secrets.ATLAS_QA_PUBLIC_API_KEY }}
+  # REALM_PRIVATE_KEY: ${{ secrets.ATLAS_QA_PRIVATE_API_KEY }}
+
+jobs:
+  run:
+    name: Build and tests with coverage
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          submodules: "recursive"
+
+      - name: Setup node version
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: npm
+
+      - name: Setup Wireit cache
+        uses: google/wireit@setup-github-actions-caching/v1
+        
+      - name: Environment setup
+        run: sudo apt-get install ccache ninja-build
+
+      - name: ccache
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          key: coverage
+          max-size: '2.0G'
+
+      - name: Docker Login
+        uses: azure/docker-login@v1
+        with:
+          login-server: ghcr.io
+          username: realm-ci
+          password: ${{ secrets.REALM_CI_GITHUB_API_KEY }}
+
+      - name: Start server
+        run: docker run --detach --env AWS_ACCESS_KEY_ID --env AWS_SECRET_ACCESS_KEY --publish 9090:9090 ghcr.io/realm/ci/mongodb-realm-test-server:${{ env.SERVER_TAG }}
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.BAAS_AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.BAAS_AWS_SECRET_ACCESS_KEY }}
```

Comment:
Running the server via docker in the job provided a +10x improvement in runtime.